lib/data/demos.dart

```diff
@@ -79,6 +81,8 @@ class GalleryDemo {
     @required this.title,
     @required this.category,
     @required this.subtitle,
+    // This parameter is required for studies.
+    this.studyId,
```

Comment:
@perclasson 
Will it break anything if we add `slug` values to studies (Shrine, etc.)? It looks fine to me, but I'm unsure.

Reply:
It should be fine to use `slug`. But it will allow us to use the studies via /demo/rally which is not intended.

We could simply update `slugToDemo` to only get the demos that are relevant:

```dart
/// Used to access code demos via slugs, used in the routes for the root Navigator.
Map<String, GalleryDemo> slugToDemo(BuildContext context) {
  final localizations = GalleryLocalizations.of(context);
  return LinkedHashMap<String, GalleryDemo>.fromIterable(
    materialDemos(localizations) +
    cupertinoDemos(localizations) +
    otherDemos(localizations),
    key: (dynamic demo) => demo.slug as String,
  );
}
```

Reply:
@guidezpl @perclasson 

Thank you for your message!

After some thought, I suggest keeping `slug` and `studyId` separate, in order to keep the code easier to understand.